scheduler/drilling-wits-depth-median/src/service.py

```diff
@@ -0,0 +1,90 @@
+import collections
+import itertools
+import statistics
+from typing import Any, Dict, List, Optional, Union
+
+from src import models
+
+
+def get_stats(data: List[dict]) -> Dict[str, Union[float, List[Optional[float]]]]:
+    # find unique keys from all data dicts
+    keys = set(itertools.chain(*data))
+
+    stats: Dict[str, Union[float, List[Optional[float]]]] = {}
+
+    for key in keys:
+        values = []
+
+        for datum in data:
+            if key not in datum:
+                continue
+
+            values.append(datum[key])
+
+        if isinstance(values[0], list):
+            # TODO: can we assume that all rows have the same length?
```

Comment:
@ville longer arrays are truncated to be equal to the length of the smallest one. Is it okay?

Reply:
For arrays of different length node does following:
Input:
arr_1: [1]
arr_2: [2,3]
As arr_1 is shorter than arr_2 it is appended with 0s
arr_1: [1,0]
arr_2: [2,3]
The result is incorrect median: [1.5, 1.5] (because there was no arr_1[1] == 0 in the real data)

Reply:
That's not good, I think it should work like this:

```
arr_1: [1]
arr_2: [2,3]

median: [1.5,3]
```